include/materials/tensor.h

```diff
@@ -0,0 +1,66 @@
+#ifndef MPM_TENSOR_H_
+#define MPM_TENSOR_H_
+
+#include <cmath>
+
+#include "Eigen/Dense"
+
+namespace mpm {
+namespace tensor {
```

Comment:
We may have to reconsider the namespace naming, none of the operations are tensorial operations. Maybe just `material` or `invariants`

Reply:
I am fine with `material`, `invariants` might seem a little too specific because this may grow bigger than just computing the `invariants`. Maybe others can comment what they think too.

Reply:
But then the name of the files would be `material` which would be the same as the current class `material`

Reply:
I vote for `material_utility`, basically, it can also contain other than invariants if future constitutive laws need it.

Reply:
Done :) unless others have better names